goredis/broker.go

```diff
@@ -0,0 +1,88 @@
+package goredis
+
+import (
+	"context"
+	"github.com/go-redis/redis/v9"
+	"github.com/marselester/gopher-celery/internal/brokertools"
+	"time"
+)
+
+// DefaultReceiveTimeout defines how many seconds the broker's Receive command
+// should block waiting for results from Redis.
+// Larger the timeout, longer the client will have to wait for Celery app to exit.
+const DefaultReceiveTimeout = 5
+
+// BrokerOption sets up a Broker.
+type BrokerOption func(*Broker)
+
+// WithBrokerPool sets Redis connection pool.
+func WithBrokerPool(pool *redis.Client) BrokerOption {
+	return func(c *Broker) {
+		c.pool = pool
+	}
+}
+
+// NewBroker creates a broker backed by Redis.
+// By default it connects to localhost.
+func NewBroker(options ...BrokerOption) *Broker {
+	br := Broker{
+		receiveTimeout: DefaultReceiveTimeout,
+	}
+	for _, opt := range options {
+		opt(&br)
+	}
+
+	if br.pool == nil {
+		// should we provide a way to pass/override redis.Options here?
+		br.pool = redis.NewClient(&redis.Options{})
```

Comment:
Would it connect to `redis://localhost` by default?

Reply:
yes that happens [here](https://github.com/go-redis/redis/blob/master/options.go#L119)